private/bufpkg/bufplugin/bufpluginconfig/config.go

```diff
@@ -214,9 +214,9 @@ func parsePluginDependency(dependency string, pluginIdentity bufpluginref.Plugin
 	if identity.Remote() != pluginIdentity.Remote() {
 		return nil, fmt.Errorf("plugin dependency %q must use same remote as plugin %q", dependency, pluginIdentity.Remote())
 	}
-	version, revisionStr, ok := strings.Cut(versionRevision, "-")
```

Comment:
This might have been a bug before, because Cut operates on the first occurrence. If the upstream semver had a pre-release component it would grab the wrong string.

Can we add a test for semver that contains a pre-release? Example:`v1.0.0-rc.1 ` 

In case we decide to change the delimiter that may be a valid semver identifier.

Reply:
Fixed: https://github.com/bufbuild/buf/pull/1270/files#diff-f714c67709cd9aa7b1bcb0fc19e7415b04b23b30974a4261ab4e7f06be52cc93R175